packages/fxa-content-server/app/scripts/views/mixins/modal-panel-mixin.js

```diff
@@ -33,8 +36,66 @@ export default {
 
     this.$el.on($.modal.AFTER_CLOSE, () => this.onAfterClose());
 
+    this._previousActiveElement = document.activeElement;
     this._boundBlockerClick = this.onBlockerClick.bind(this);
     $('.blocker').on('click', this._boundBlockerClick);
+
+    /**
+     * Trap keyboard focus when modal opens.
+     */
+    const modal = this.$el[0];
+    modal.setAttribute('tabindex', '1');
+    modal.focus();
+
+    const focusableElementsNodeList = modal.querySelectorAll(
+      '#file, #camera, button, input[type="radio"]'
+    );
+    const focusableElements = Array.prototype.slice.call(
+      focusableElementsNodeList
+    );
+
+    this._keyDownListener = this.keyDownListener.bind(null, focusableElements);
+    modal.addEventListener('keydown', this._keyDownListener);
+  },
+
+  keyDownListener(focusableElements, event) {
+    function getNextSelectable(element = focusableElements[0]) {
+      const nextIndex = focusableElements.indexOf(element) + 1;
+      // return first element if the next element doesn't exist
+      return focusableElements[nextIndex]
+        ? focusableElements[nextIndex]
+        : focusableElements[0];
+    }
+
+    function getPreviousSelectable(
+      element = focusableElements[focusableElements.length - 1]
+    ) {
+      const previousIndex = focusableElements.indexOf(element) - 1;
+      // return last element if the previous element doesn't exist
+      return focusableElements[previousIndex]
+        ? focusableElements[previousIndex]
+        : focusableElements[focusableElements.length - 1];
+    }
+
+    //avoid IME composition keydown events
+    //ref: https://developer.mozilla.org/docs/Web/Events/keydown#Notes
+    if (event.isComposing || event.keyCode === 229) return;
```

Comment:
Can you add `229` to [key-codes.js](https://github.com/mozilla/fxa/blob/0bbac2ba3666749309d9bfc03448fc38a1b21331/packages/fxa-content-server/app/scripts/lib/key-codes.js#L4) and use a named constant here?

Also, can you put the `return` on its own line and add some block`{}`?

I thought we had a lint rule about that.

Reply:
Done. Haha, developer preference. I don't even use semicolons on personal projects -le gasp-